.github/workflows/release.yml

```diff
@@ -1,58 +1,113 @@
-name: Release conda-store-ui
+name: "Build and maybe release conda-store-ui"
 
 on:
-  # we want to make a release whenever a new tag is created
+  release:
+    types: [published]
   push:
-    tags:
-      - "*"
+    branches: [main]
+    tags: ["*"]
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
 
-jobs:
-  call-build:
-    uses: conda-incubator/conda-store-ui/.github/workflows/build.yml@main
+env:
+  FORCE_COLOR: "1"
+  PACKAGE_FILE: "conda-store-ui.tgz"
 
-  make-release:
+jobs:
+  # always build and verify
+  build-application:
+    name: "Build conda-store-ui"
     runs-on: ubuntu-latest
-    # ensure that the artifacts are available from the build job
-    needs: call-build
 
     steps:
       - name: "Checkout repository 🛎"
         uses: actions/checkout@v4
 
-      # Setup .npmrc file to publish to npm
       - name: "Set up Node.js 🧶"
         uses: actions/setup-node@v4
         with:
-          node-version: 18
-          registry-url: "https://registry.npmjs.org"
-          scope: "@conda-store-ui"
+          node-version: "20.x"
+          cache: "yarn"
 
       - name: "Install dependencies 📦"
         run: yarn
 
-      - name: "Lint code 🔎"
+      - name: "Lint code 🔍"
         run: yarn eslint:check
 
-      - name: "Download webpack bundle 📦"
+      - name: "Build application 🏗"
+        run: yarn run build
+
+      - name: "Build application artifacts 🏗"
+        # ensure we use the prod target to minimise assets
+        run: yarn run webpack:prod bundle
+
+      - name: "Generate package tarball 📦"
+        run: yarn pack --filename ${{ env.PACKAGE_FILE }}
+
+      - name: "Upload package tarball 📤"
+        uses: actions/upload-artifact@v4
+        with:
+          name: conda-store-ui-package
+          path: ${{ env.PACKAGE_FILE }}
+
+  release-to-npmjs:
+    name: "Release conda-store-ui to npmjs 📦"
+    runs-on: ubuntu-latest
+    needs: build-application
+    # needed for attestations
+    permissions:
+      id-token: write
+      attestations: write
+      contents: read
+
+    steps:
+      - name: "Checkout repository 🛎"
+        uses: actions/checkout@v4
+
+      # Setup .npmrc file to publish to npm
+      - name: "Set up Node.js 🧶"
```

Comment:
yes so that we have an `.npmrc` and can handle authentication to npmjs

Reply:
It does not magically work - I am using the `registry-url` here and it still fails to pick up the correct credentials.

See https://github.com/actions/setup-node/issues/763
which I tried and check if it was resolved, but no.

Reply:
To be clear, I was agreeing with you here that this step is necessary. It's just sad to me that these two things are not at all clearly related. It's side effect dependencies.

I think a comment like the following would be helpful to people later:

```
# Without this parameter, `npm publish` will not be able to get the API token from the environment variable `NODE_AUTH_TOKEN`
# (Why? Unsure but it seems that registry-url causes .npmrc to be saved which somehow causes npm publish to read from NODE_AUTH_TOKEN) 
registry-url: "https://registry.npmjs.org"
```

Reply:
Also I don't understand why we keep linking to actions/setup-node#763. That issue is irrelevant. It's only relevant if you're using the GitHub package registry, which we're not using.

Reply:
But it is not, that is what I am trying to explain.  
Using the set-node action with a registry-url should configure it so that the authentication is properly setup for publication to whichever registry. 
But it is not, hence I needed to add, a while back, the additional step to set the proper context. 

That issue is relevant as it's the only issue where I found useful steps to properly set the context and authentication. Hence why I left it as a comment. 
I tried to remove the additional context-setup again this time to see if that was fixed but the authentication was still failing, so could not remove that step. I can remove the comment linking to that issue if this is too confusing but it's helpful to me as a reference.